plugins/modules/dns_auth_zone.py

```diff
@@ -0,0 +1,2316 @@
+                        - "Defaults to empty."
+                    type: str
+                    returned: Always
+                refresh:
+                    description:
+                        - "Optional. ZoneAuthority refresh."
+                        - "Defaults to 10800."
+                    type: int
+                    returned: Always
+                retry:
+                    description:
+                        - "Optional. ZoneAuthority retry."
+                        - "Defaults to 3600."
+                    type: int
+                    returned: Always
+                rname:
+                    description:
+                        - "Optional. ZoneAuthority rname."
+                        - "Defaults to empty."
+                    type: str
+                    returned: Always
+                use_default_mname:
+                    description:
+                        - "Optional. Use default value for master name server."
+                        - "Defaults to true."
+                    type: bool
+                    returned: Always
+"""  # noqa: E501
+
+from ansible_collections.infoblox.bloxone.plugins.module_utils.modules import BloxoneAnsibleModule
+
+try:
+    from bloxone_client import ApiException, NotFoundException
+    from dns_config import AuthZone, AuthZoneApi
+except ImportError:
+    pass  # Handled by BloxoneAnsibleModule
+
+
+class AuthZoneModule(BloxoneAnsibleModule):
+    def __init__(self, *args, **kwargs):
+        super(AuthZoneModule, self).__init__(*args, **kwargs)
+
+        exclude = ["state", "csp_url", "api_key", "id"]
+        self._payload_params = {k: v for k, v in self.params.items() if v is not None and k not in exclude}
+        self._payload = AuthZone.from_dict(self._payload_params)
+        self._existing = None
+
+    @property
+    def existing(self):
+        return self._existing
+
+    @existing.setter
+    def existing(self, value):
+        self._existing = value
+
+    @property
+    def payload_params(self):
+        return self._payload_params
+
+    @property
+    def payload(self):
+        return self._payload
+
+    def payload_changed(self):
+        if self.existing is None:
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = AuthZoneApi(self.client).read(self.params["id"], inherit="full")
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "absent":
+                    return None
+                raise e
+        else:
+            filter = f"fqdn=='{self.params['fqdn']}' and primary_type=='{self.params['primary_type']}'"
```

Comment:
i believe `fqdn` alone should be enough here to uniquely identify the zone.

Reply:
Fixed